pegasus/sites.v3/code.org/views/display_language_lightbox.haml

```diff
@@ -37,7 +37,35 @@
       $.getJSON( "/v2/client-location", function( data ) {
         country_code = data['country_code'];
         if(country_code != 'US') {
-          $('#selectLanguage').modal('show')
+          $('#selectLanguage').modal('show');
+          $('#selectLanguage').on('shown.bs.modal', function () {
+            var focusableEls = $(this).find('a[href], button, textarea, input[type="text"], input[type="radio"], input[type="checkbox"], select');
+            var firstFocusableEl = focusableEls.first()[0];
+            var lastFocusableEl = focusableEls.last()[0];
+            var KEYCODE_TAB = 9;
+
+            firstFocusableEl.focus();
+
+            $(this).on('keydown', function(e) {
+              var isTabPressed = (e.key === 'Tab' || e.keyCode === KEYCODE_TAB);
+
+              if (!isTabPressed) { 
+                return; 
+              }
+
+              if (e.shiftKey) {
+                if (document.activeElement === firstFocusableEl) {
+                  lastFocusableEl.focus();
+                  e.preventDefault();
+                }
+              } else {
+                if (document.activeElement === lastFocusableEl) {
+                  firstFocusableEl.focus();
+                  e.preventDefault();
+                }
```

Comment:
I would appreciate it if there were a few inline comments for this code explaining the motivation:
1. Have a summary at the beginning about what a focus trap is and why we are doing it. Just a few sentences should be enough. Assume the engineer reading it doesn't know what a focus trap is.
1. Have some inline comments at key functionality explaining  what is happening, and or the motivation. For example: "If the user trys to tab past the last focusable element in the modal, force the focus to the first element in the modal."